cmake/SanitizerOptions.cmake

```diff
@@ -0,0 +1,34 @@
+if (ENABLE_SANITIZERS)
+  set(ENABLE_ADDRESS_SANITIZER ON)
+  set(ENABLE_UNDEFINED_SANITIZER ON)
+endif()
+
+# Inlclude build macros for updating configuration variables
+include(HandleLibcxxFlags)
+
+set (SANITIZER_FLAGS_LIST "")
+set (SANITIZER_EXE_LINKER_FLAGS_LIST "")
+set (SANITIZER_SHARED_LINKER_FLAGS_LIST "")
+
+if(ENABLE_ADDRESS_SANITIZER)
+  message(STATUS "Enabling address sanitizer (ASan).")
+  set(CMAKE_REQUIRED_FLAGS "-fsanitize=address")
```

Comment:
Need this trick for `-fsanitize=address`. Without it the flag was not found for some reason. The idea is from [here](https://github.com/arsenm/sanitizers-cmake/blob/master/cmake/sanitize-helpers.cmake#L120-L121)

Interesting that `-fsanitize=undefined` is ok without this trick

Reply:
Doesn't this overwrite any previous value of `CMAKE_REQUIRED_FLAGS`?

Reply:
That's a good point. I can't find any sensible doc regarding this flag, only came across a note saying "CMAKE_REQUIRED_FLAGS is explicitly designed for temporary flags used in CMake's various check_<something> macros". **But** in order to be on the safe side I'll better save old values and restore it afterwards

Reply:
Done